sections/merge-accounts/merge/MergeActionBox.tsx

```diff
@@ -0,0 +1,411 @@
+		},
+		[isAppReady, properSourceAccountAddress, entryIDs, sourceAccountAddressInputError, synthetixjs]
+	);
+
+	// gas
+	useEffect(() => {
+		let isMounted = true;
+		(async () => {
+			try {
+				setError(null);
+				const data: any[] | null = getMergeTxData({});
+				if (!data) return;
+				const [contract, method, args] = data;
+				const gasEstimate = await getGasEstimateForTransaction(args, contract.estimateGas[method]);
+				if (isMounted) setGasLimitEstimate(getNormalizedGasLimit(Number(gasEstimate)));
+			} catch (error) {
+				// console.error(error);
+				if (isMounted) setGasLimitEstimate(null);
+			}
+		})();
+		return () => {
+			isMounted = false;
+		};
+	}, [getMergeTxData, synthetixjs]);
+
+	// load any nominated account address
+	useEffect(() => {
+		if (!isAppReady) return;
+		const {
+			contracts: { RewardEscrowV2 },
+		} = synthetixjs!;
+		if (!properSourceAccountAddress) return;
+
+		let isMounted = true;
+		const unsubs = [
+			() => {
+				isMounted = false;
+			},
+		];
+
+		const load = async () => {
+			const nominatedAccountAddress = await RewardEscrowV2.nominatedReceiver(
+				properSourceAccountAddress
+			);
+			if (isMounted) {
+				if (ethers.constants.AddressZero !== nominatedAccountAddress) {
+					setNominatedAccountAddress(nominatedAccountAddress);
+				}
+				setEntryIDs(entryIDs);
+			}
+		};
+
+		const subscribe = () => {
+			const contractEvent = RewardEscrowV2.filters.NominateAccountToMerge(
+				properSourceAccountAddress
+			);
+			const onContractEvent = async (src: string, dest: string) => {
+				if (src === destinationAccountAddress) {
+					setNominatedAccountAddress(dest);
+				}
+			};
+			RewardEscrowV2.on(contractEvent, onContractEvent);
+			unsubs.push(() => {
+				RewardEscrowV2.off(contractEvent, onContractEvent);
+			});
+		};
+
+		load();
+		subscribe();
+		return () => {
+			unsubs.forEach((unsub) => unsub());
+		};
+	}, [destinationAccountAddress, properSourceAccountAddress, entryIDs, isAppReady, synthetixjs]);
+
+	// load nominated account address
+	useEffect(() => {
+		if (!isAppReady) return;
+		const {
+			contracts: { RewardEscrowV2 },
+		} = synthetixjs!;
+		if (!properSourceAccountAddress) return;
+
+		let isMounted = true;
+		const unsubs = [
+			() => {
+				isMounted = false;
+			},
+		];
+
+		const load = async () => {
+			const numVestingEntries = await RewardEscrowV2.numVestingEntries(properSourceAccountAddress);
+			const entryIDs = numVestingEntries.eq(0)
+				? []
+				: await RewardEscrowV2.getAccountVestingEntryIDs(
+						properSourceAccountAddress,
+						0,
+						numVestingEntries
+				  );
+
+			if (isMounted) {
+				setEntryIDs(entryIDs);
+			}
+		};
+
+		load();
+
+		return () => {
+			unsubs.forEach((unsub) => unsub());
+		};
+	}, [properSourceAccountAddress, isAppReady, synthetixjs]);
+
+	// funcs
+
+	const onEnterAddress = (e: any) => setSourceAccountAddress((e.target.value ?? '').trim());
+
+	const connectOrMerge = async () => {
```

Comment:
again you can save like half of this code by using `useSynthetixTxn` mutation

Reply:
done